src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -252,6 +253,7 @@ Status ICDManagementServer::RegisterClient(CommandHandler * commandObj, const Co
     FabricIndex fabricIndex            = commandObj->GetAccessingFabricIndex();
     NodeId nodeId                      = commandData.checkInNodeID;
     uint64_t monitoredSubject          = commandData.monitoredSubject;
+    ClientTypeEnum clientType          = commandData.clientType;
```

Comment:
Something here needs to check that this is a valid ClientTypeEnum value, no?

Reply:
done